awsbestpractices/missing-pagination/MissingPagination.java

```diff
@@ -0,0 +1,507 @@
+import com.amazonaws.services.dynamodbv2.model.AttributeValueUpdate;
+import com.amazonaws.services.dynamodbv2.model.ComparisonOperator;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.amazonaws.services.dynamodbv2.model.GetItemRequest;
+import com.amazonaws.services.dynamodbv2.model.GetItemResult;
+import com.amazonaws.services.dynamodbv2.model.PutItemRequest;
+import com.amazonaws.services.dynamodbv2.model.PutItemResult;
+import com.amazonaws.services.dynamodbv2.model.ReturnValue;
+import com.amazonaws.services.dynamodbv2.model.ScanRequest;
+import com.amazonaws.services.dynamodbv2.model.ScanResult;
+import com.amazonaws.services.dynamodbv2.model.UpdateItemRequest;
+import com.amazonaws.services.dynamodbv2.model.UpdateItemResult;
+
+
+public class MissingPagination {
+    DynamoDB dynamoClient;
+    DynamoDBMapper mapper;
+    private AmazonCloudFormationClient cloudformation;
+
+    // Missing
+    public Set<String> missingPagination_1(){
+
+        this.tableName = EXPIRY_TABLE;
+
+        Set<String> expiredStores = new HashSet<String>();
+
+        Condition stateCondition = new Condition().
+                withComparisonOperator(ComparisonOperator.EQ.toString()).
+                withAttributeValueList(new AttributeValue().withS("P"));
+
+        Map<String, Condition> conditions = new HashMap<String,Condition>();
+        conditions.put("STATUS",stateCondition);
+
+        ScanRequest scanRequest = new ScanRequest()
+                .withTableName(tableName)
+                .withScanFilter(conditions)
+                .withAttributesToGet(Arrays.asList("STORE_ID"));
+
+
+        ScanResult result = dynamoClient.scan(scanRequest);
+        if(result==null || result.getScannedCount()==0){
+            logger.info("zero scanned");
+        }
+
+        logger.info("Set of expired stores found");
+
+        for (Map<String, AttributeValue> item : result.getItems()) {
+
+            String storeID = item.get("STORE_ID").getS();
+            expiredStores.add(storeID);
+
+        }
+
+        return expiredStores;
+
+    }
+
+    // Missing
+    // Adapted from
+    // https://code.amazon.com/packages/AwsGlueMlArtifactsDeletionLambda/blobs/0b683300e7a9e42b10a7392a52a9e7f3d64308b2/--/src/com/amazon/glue/ml/lambda/artifacts/deletion/OldMLTransformArtifactsDeletion.java#L128-L133
+    public Set<String> missingPagination_2(AmazonS3 s3Client, TransformCleanupEntry transformCleanupEntry) {
+        MLModelArtifact mlModelArtifact = new MLModelArtifact(transformCleanupEntry.getAccountId(), transformCleanupEntry.getTransformId(),
+                "dummyFileName");
+
+        String bucketName = mlModelArtifact.bucketName(stage, region);
+        String key = String.format("%s/%s/", transformCleanupEntry.getAccountId(), transformCleanupEntry.getTransformId());
+        List<S3ObjectSummary> s3ObjectSummaries = s3Client.listObjects(bucketName, key).getObjectSummaries();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses an outdated API. [ListObjectsV2 is the revised List Objects API, and we recommend you use this revised API for new application developments](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html).